ulib/fs/FStar_HyperStack_ST.fs

```diff
@@ -58,27 +58,33 @@ let witness_region _ = ()
 let witness_hsref _ = ()
 type erid = rid
 
+type ('a, 'rel) mreference = ('a, 'rel) FStar_Monotonic_HyperStack.mreference
+type ('a, 'rel) mref = ('a, 'rel) FStar_Monotonic_HyperStack.mref
+type 'a reference = ('a, unit) mreference
+type 'a ref = ('a, unit) mref
+type ('a, 'b) m_rref = ('a, 'b) mref
```

Comment:
I've took this file from ulib/ml but it seems to be seriously out of sync with FStar.HyperStack.ST. When I take those five type definitions from the extracted version of FStar.HyperStack.ST, the Monotonic.Map.fst example extracts fine. In fact, I can even enable entire FStar.Monotonic.Map.fst in ulib. 

It seems to work in OCaml because the mref definition below simply ignores 'b that extracted FStar.Monotonic.Map fills with "unit". 

@nikswamy Does it make sense to update the manual implementation? What about OCaml version?